pkg/storage/db.go

```diff
@@ -1109,34 +1093,41 @@ func (db *DB) StoreAccountOwner(ctx context.Context, accountOwner domain.Account
 
 func (db *DB) storeAccountOwner(ctx context.Context, accountOwner domain.AccountOwner, tx *sql.Tx) error {
 
-	sqlStatement := fmt.Sprintf(`REPLACE INTO aws_account (account) VALUES($1)`)
+	sqlStatement := fmt.Sprintf(`
+				INSERT INTO aws_account (account)
+				VALUES ($1)
+				ON CONFLICT DO NOTHING
+				`)
 	var err error
 	if _, err = tx.ExecContext(ctx, sqlStatement, accountOwner.AccountID); err != nil {
 		return err
 	}
-	// Insert or update details of account owner in "person" table
+	// Insert or update details of account owner
 	if _, err = tx.ExecContext(ctx, insertPersonQuery, accountOwner.Owner.Login, accountOwner.Owner.Email, accountOwner.Owner.Name, accountOwner.Owner.Valid); err != nil {
 		return err
 	}
 
 	sqlStatement = fmt.Sprintf(`SELECT id FROM person WHERE login=$1`)
-	row, err := tx.QueryContext(ctx, sqlStatement, accountOwner.Owner.Login)
-	if err != nil {
+	row := tx.QueryRowContext(ctx, sqlStatement, accountOwner.Owner.Login)
+	var personID int
+	if err := row.Scan(&personID); err != nil {
 		return err
 	}
-	var id int
-	row.Next()
-	if err := row.Scan(&id); err != nil {
+
+	sqlStatement = fmt.Sprintf(`SELECT id FROM aws_account WHERE account=$1`)
+	row = tx.QueryRowContext(ctx, sqlStatement, accountOwner.AccountID)
+	var accountID int
+	if err := row.Scan(&accountID); err != nil {
 		return err
 	}
 
-	sqlStatement = fmt.Sprintf(insertAccountOwnerQuery, id)
-	if _, err := tx.ExecContext(ctx, sqlStatement, accountOwner.AccountID); err != nil {
+	sqlStatement = fmt.Sprintf(`INSERT INTO account_owner VALUES %d, %d`, personID, accountID)
```

Comment:
Should this query also handle updates to account_owner? for example, if the account_id already exists in the table but there is a new person_id assigned to it

Reply:
Yes. When there is another pair of accoun_id and person_id from input, the pair will be inserted into the table if the account_id already exists. The unique key in the table is (account_id, person_id) so the table will be inserted or updated, except there is input with existing account_id and person_id pair.

Reply:
No. This is because values in aws_account_id column should be unique according to account_owner table. I am fixing the query. Thank you for your question.

Reply:
Done